src/Transactions/TransactionDetails/TransactionInformation/SourceValue/useSourceValue.js

```diff
@@ -0,0 +1,34 @@
+import { useQuery } from 'react-query';
+
+import { useOkapiKy } from '@folio/stripes/core';
+import {
+  LINES_API,
+  INVOICES_API,
+} from '@folio/stripes-acq-components';
+
+import { TRANSACTION_SOURCE } from '../../../constants';
+
+const queryFnsMap = {
+  [TRANSACTION_SOURCE.invoice]: async (ky, transaction) => {
+    const { folioInvoiceNo } = await ky(`${INVOICES_API}/${transaction.sourceInvoiceId}`).json();
```

Comment:
have you verified this code when there is 404 coming from the back-end?

Reply:
Yes, in that case, returns `{ isLoading: false, date: undefied }`. It handles [here](https://github.com/folio-org/ui-finance/pull/497/files#diff-a2a4a1b5bbec426d382e173c9e69a3e5671875b63bf5695ddffd678f373cc63aR23)

Reply:
personally I see that you solution not optimal. You have some logic inside useSourceValue , that sounds like it returns source value, and at the same time you have at least 2 extra OR at the place where you've provided the link above. IMHO, if you create a hook, like a solid and single responsible for returning source value - all logic could live there

Reply:
I still looks confusing, you a lot of OR's in the place you've provided the link and no single responsibility to generate link